pkg/ipamd/ipamd.go

```diff
@@ -636,6 +636,10 @@ func (c *IPAMContext) updateIPPoolIfRequired(ctx context.Context) {
 
 	if c.isDatastorePoolTooLow() {
 		c.increaseDatastorePool(ctx)
+		// Speeds up ENI scale up
+		for c.isDatastorePoolTooLow() {
```

Comment:
Let's wait for https://github.com/aws/amazon-vpc-cni-k8s/pull/2745 to merge, as we should be able to make this a bit cleaner, i.e. something like:
```
poolTooLow, stats := c.isDatastorePoolTooLow()
if poolTooLow {
    for stillTooLow := poolTooLow; stillTooLow; stillTooLow, _ = c.isDatastorePoolTooLow(nil) {
        c.increaseDatastorePool(ctx)
    }
} else if c.isDatastorePoolTooHigh(stats) {
    ...
}
```